porthole/components.py

```diff
@@ -170,6 +172,63 @@ def create_worksheet_from_query(self, cm, sheet_name, query=None, sql=None, quer
         self.make_worksheet(sheet_name=sheet_name, query_results=results, **worksheet_kwargs)
 
 
+class ReportReadWriter(Loggable):
+
+    def __init__(self, report_filename: str
+                    , read_file_path: Optional[str]=None
+                    , write_file_path: Optional[str]=None
+                    , log_to=None):
+        self.report_filename = report_filename
+        if not read_file_path:
+            self.read_file_path = config['Default'].get('base_file_path')
+        if not write_file_path:
+            self.write_file_path = self.read_file_path
+        self.read_location = os.path.join(self.read_file_path, self.report_filename)
+        self.write_location = os.path.join(self.write_file_path, self.report_filename)
+        self.record_count = 0
+        super().__init__(log_to=log_to if log_to is not None else [])
+
+    def update_sheet_on_report_workbook(self, cm: ConnectionManager
+                                        , sheet_name: str
+                                        , query: Optional[dict]=None
+                                        , sql: Optional[str]=None
+                                        , query_kwargs: Optional[dict]=None
+                                        , save_as: Optional[str]=None) -> None:
+        if query is None:
+            query = {}
+        if query_kwargs is None:
+            query_kwargs = {}
+        if save_as:
+            self.write_location = os.path.join(self.write_file_path, save_as)
+        
+        results = self.execute_query(cm=cm, query=query, sql=sql, **query_kwargs)
+        formatted_results = results.as_list_of_tuples()
+        wb = WorkbookEditor(workbook_filename=self.read_location)
+        wb.replace_sheet_contents(sheet_name=sheet_name, data_rows=formatted_results)
+        wb.save_workbook(save_as=self.write_location)
+
+    def execute_query(self, cm: ConnectionManager
```

Comment:
Have you considered using QueryExecutor here instead? You could cut out most of the moving parts here, as long as you make sure to handle the increment counter piece separately.

Reply:
This method is almost a direct copy of the same method in `ReportWriter()`.  I felt it was valuable these methods mirror each other.  I'm open to changing both, if you think it is worth it.